tools/Customization/DevHome.Customization/ViewModels/VirtualizationFeatureManagementViewModel.cs

```diff
@@ -0,0 +1,285 @@
+
+    public ObservableCollection<Breadcrumb> Breadcrumbs { get; }
+
+    public ObservableCollection<WindowsOptionalFeatureState> Features { get; } = new();
+
+    public bool HasFeatureChanges => _isUserAdministrator && FeaturesLoaded && Features.Any(f => f.HasChanged);
+
+    public bool CanDismissNotifications => _isUserAdministrator;
+
+    private bool _restartNeeded;
+
+    private bool _isFirstLoad;
+
+    public VirtualizationFeatureManagementViewModel(Window window)
+    {
+        _stringResource = new StringResource("DevHome.Customization.pri", "DevHome.Customization/Resources");
+        _window = window;
+        _isFirstLoad = true;
+
+        Breadcrumbs =
+        [
+            new(_stringResource.GetLocalized("MainPage_Header"), typeof(MainPageViewModel).FullName!),
+            new(_stringResource.GetLocalized("VirtualizationFeatureManagement_Header"), typeof(VirtualizationFeatureManagementViewModel).FullName!)
+        ];
+
+        LoadFeaturesCommand = new AsyncRelayCommand(LoadFeaturesAsync);
+        LoadFeaturesCommand.PropertyChanged += async (s, e) =>
+        {
+            if (e.PropertyName == nameof(LoadFeaturesCommand.IsRunning))
+            {
+                await OnFeaturesChanged();
+            }
+        };
+
+        ApplyChangesCommand = new AsyncRelayCommand(ApplyChangesAsync);
+        ApplyChangesCommand.PropertyChanged += async (s, e) =>
+        {
+            if (e.PropertyName == nameof(ApplyChangesCommand.IsRunning))
+            {
+                await OnFeaturesChanged();
+            }
+        };
+
+        _modifyFeaturesDialog = new ModifyFeaturesDialog(ApplyChangesCommand)
+        {
+            XamlRoot = _window.Content.XamlRoot,
+        };
+    }
+
+    internal async Task Initialize(StackedNotificationsBehavior notificationQueue)
+    {
+        _notificationQueue = notificationQueue;
+
+        var loadingTask = LoadFeaturesCommand.ExecuteAsync(null);
+
+        if (!_isUserAdministrator)
+        {
+            await _window.DispatcherQueue.EnqueueAsync(ShowNonAdminUserNotification);
+        }
+
+        if (_restartNeeded)
+        {
+            await _window.DispatcherQueue.EnqueueAsync(ShowRestartNotification);
+        }
+
+        await loadingTask;
+    }
+
+    internal void Uninitialize()
+    {
+        _notificationQueue = null;
+    }
+
+    private async void FeatureState_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(WindowsOptionalFeatureState.IsEnabled))
+        {
+            await OnFeaturesChanged();
+        }
+    }
+
+    private async Task LoadFeaturesAsync()
+    {
+        var tempFeatures = new ObservableCollection<WindowsOptionalFeatureState>();
+
+        await Task.Run(() =>
+        {
+            foreach (var featureName in WindowsOptionalFeatures.VirtualMachineFeatures)
+            {
+                var feature = ManagementInfrastructureHelper.GetWindowsFeatureDetails(featureName);

```

Comment:
LINQ can grab all features that aren't null and where the feature is available.  If that query is added to the end of `GetWindowsFeatureDetails` then the `if` can be removed.

Reply:
I'll defer on this one since I'm not sure about other use-cases as to whether having those features without availability returned is important/interesting.